### PR TITLE
[PUBDEV-7082] Expose Location of Chunks in REST API

### PR DESCRIPTION
**Pull request by mn-mikke**

### CODE REVIEW

File: h2o-core/src/main/java/water/api/schemas3/FrameChunksV3.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    public static class FrameChunkV3 extends SchemaV3<Iced, FrameChunkV3> {
+
+        @API(help="An identifier unique in scope of a given frame", direction=API.Direction.OUTPUT)
+        public int chunk_id;
+
+        @API(help="Number of rows represented byt the chunk", direction=API.Direction.OUTPUT)
+        public int row_count;
+        
+        @API(help="Description of H2O node where the chunk is located", direction=API.Direction.OUTPUT)
+        public CloudV3.NodeV3 location;
```
Comment (michalkurka) [inline h2o-core/src/main/java/water/api/schemas3/FrameChunksV3.java]: NodeV3 has quite a bit of information in it, this info will be replicated many times as many chunks will reside on the same node. I would use node index instead. WDYT?

File: h2o-core/src/test/java/water/api/APIThrPriorTest.java [hunk 2]
```diff
@@ -97,7 +97,7 @@
       vec.sigma();
       Assert.assertNotNull(DKV.get(rskey)); // Rollups on my zeros not computed yet
       serve("/3/Frames/iris.hex", null,200); // Rollups already done at parse, but gets ChunkSummary
-
+      serve("/3/FrameChunks/iris.hex", null,200);
```
Comment (michalkurka) [inline h2o-core/src/test/java/water/api/APIThrPriorTest.java]: doesn't really belong here imho

File: h2o-core/src/main/java/water/api/schemas3/FrameChunksV3.java (hunk 1 above)
Comment (mn-mikke) [reply h2o-core/src/main/java/water/api/schemas3/FrameChunksV3.java]: I like your idea, but I can't find anything like this on `NodeV3` is the node index equal to the position in `CloudeV3.nodes`?

File: h2o-core/src/test/java/water/api/APIThrPriorTest.java (hunk 2 above)
Comment (mn-mikke) [reply h2o-core/src/test/java/water/api/APIThrPriorTest.java]: ok, will remove it.

File: h2o-core/src/main/java/water/api/schemas3/FrameChunksV3.java (hunk 1 above)
Comment (michalkurka) [reply h2o-core/src/main/java/water/api/schemas3/FrameChunksV3.java]: you can call `key.home().index()`

  // index of this node in the current cloud... can change at the next cloud.
  public int index() { return H2O.CLOUD.nidx(this); }

Comment (mn-mikke) [reply h2o-core/src/main/java/water/api/schemas3/FrameChunksV3.java]: I mentioned the position in `CloudV3.nodes` in order to be able to correctly map the node index to the correct `NodeV3` instance within the array. [It seems that Sparkling Water already relies on the equality of the node order within `CloudV3.nodes` and the node index.](https://github.com/h2oai/sparkling-water/blob/features/dynamic-allocation/core/src/main/scala/org/apache/spark/h2o/utils/H2OContextRestAPIUtils.scala#L102)

Comment (michalkurka) [reply h2o-core/src/main/java/water/api/schemas3/FrameChunksV3.java]: Yes, the order is given once the cloud is formed. You can rely on that.

Comment (mn-mikke) [reply h2o-core/src/main/java/water/api/schemas3/FrameChunksV3.java]: Cool, thanks!

**Comment by mn-mikke** [comment]
@jakubhava @michalkurka The call of this new REST API was causing failure of the whole H2O node. I made same changes to avoid that. Please can you have a second look?

File: h2o-core/src/main/java/water/api/schemas3/FrameChunksV3.java [hunk 3]
```diff
@@ -5,9 +5,9 @@
 import water.api.API;
 import water.fvec.Vec;
 
-public class FrameChunksV3 extends SchemaV3<Frame, FrameChunksV3> {
+public class FrameChunksV3 extends SchemaV3<Iced, FrameChunksV3> {
```
Comment (michalkurka) [inline h2o-core/src/main/java/water/api/schemas3/FrameChunksV3.java]: I think this is a bit unusual but I don't see any issues with it

**Comment by michalkurka** [comment]
@mn-mikke I don't see any issue - I think it is good to merge

File: h2o-core/src/main/java/water/api/schemas3/FrameChunksV3.java (hunk 3 above)
Comment (mn-mikke) [reply h2o-core/src/main/java/water/api/schemas3/FrameChunksV3.java]: There is a problem that `Frame` doesn't have a parameterless constructor and [this line](https://github.com/h2oai/h2o-3/blob/b9b0954b45c956d6d3d08537ada3543c85efa557/h2o-core/src/main/java/water/api/Schema.java#L167) will cause H2O node failure. I guess that's the reason why the `Frames` class was introduced.

Regarding the usage of `Iced`, I've gotten inspired in [`FindV3`](https://github.com/h2oai/h2o-3/blob/43f8ab952a69a8bc9484bd0ffac909b6e3e820ca/h2o-core/src/main/java/water/api/schemas3/FindV3.java#L7).

**Comment by Pscheidl** [comment]
The tests are OK. Failures completely unrelated. :heavy_check_mark:

**Comment by mn-mikke** [comment]
Thank guys for your reviews!

**Pull request merged**